odoo/tools/misc.py

```diff
@@ -1148,6 +1148,9 @@ def difference_update(self, elems):
     def __repr__(self):
         return f'{type(self).__name__}({list(self)!r})'
 
+    def intersection(self, *others):
+        return reduce(self.__and__, others)
```

Comment:
Good idea to use reduce here with __and__ (which allows every type of Iterable). Sadly this version is not correct: 
```
OrderedSet([1,2,3]).intersection([5, 1,2], [2,3])

=> Cell In[3], line 25, in OrderedSet.intersection(self, *others)
     24 def intersection(self, *others):
---> 25     return reduce(self.__and__, others)

TypeError: Set.__and__() takes 2 positional arguments but 3 were given
```
It is because `self.__and__` is the bounded method of `OrderedSet.__and__`.

Here is the correct implementation with reduce:
```python
 def intersection(self, *others):
        return reduce(OrderedSet.__and__, others, self)
```